hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/types/HazelcastTypeCoercion.java

```diff
@@ -111,11 +114,6 @@ private static SqlNode cast(SqlNode node, SqlDataTypeSpec targetTypeSpec) {
         }
     }
 
-    @Override
-    protected boolean coerceColumnType(SqlValidatorScope scope, SqlNodeList nodeList, int index, RelDataType targetType) {
-        throw new UnsupportedOperationException("Should not be called");
```

Comment:
So far the strategy for implementing any functionality in this class was to not to delegate to super class and deliberately copy required code if needed (similar comment for `inOperationCoercion()`).

Reply:
We don't need to do any change in the superclass code. The approach until now wasn't to copy the whole code, but to avoid the need for calling this method by implementing `checkOperandTypes` in the operators. But Calcite handles it differently for `IN` operator and doesn't call the method for `HazelcastInOperator`. I'm not too keen on hacking this more :)